C++/API/API.cpp

```diff
@@ -170,231 +153,208 @@ void iSENSE::clear_data(void)
     return false;
-  }
-  else if(email.empty() || password.empty())
-  {
-    cerr << "Please set an email & password for this project.\n";
+  } else if (email.empty() || password.empty()) {
+    std::cerr << "\nPlease set an email & password for this project.\n";
     return false;
   }
 
   // If we get here, an email and password have been set, so do a GET using
   // the email & password.
-  get_UserURL = devURL + "/users/myInfo?email=" + email + "&password=" + password;
+  get_UserURL = devURL + "/users/myInfo?email=" +
+                email + "&password=" + password;
 
   // This project will try using CURL to make a basic GET request to rSENSE
   CURL *curl = curl_easy_init();          // cURL object
-  long http_code;                         // HTTP status code
+  long http_code = 0;                     // HTTP status code
 
-  if(curl)
-  {
-    // Set the GET URL, in this case the one be created above using the user's
-    // email & password.
+  if (curl) {
+    // Set the GET URL
     curl_easy_setopt(curl, CURLOPT_URL, get_UserURL.c_str());
 
-    // Disable libcURL's desire to output to the screen by giving it a
-    // write function which basically does nothing but returns the size of what
-    // would have been outputted.
+    // Stop libcURL from outputting to stdio.
     curl_easy_setopt(curl, CURLOPT_WRITEFUNCTION, suppress_output);
 
-    // Perform the request
-    curl_easy_perform(curl);
-
-    // We can actually get the HTTP response code from cURL, so let's do so to check for errors.
-    http_code = 0;
+    curl_easy_perform(curl);    // Perform the request
 
     // This will put the HTTP response code into the "http_code" variable.
     curl_easy_getinfo(curl, CURLINFO_RESPONSE_CODE, &http_code);
+  }
 
-    /*
-     *      The iSENSE API gives us two response codes to check against:
-     *      Success: 200 OK
-     *      Failure: 401 Unauthorized
-     */
-
-    if(http_code == 200)
-    {
-      // Clean up cURL
-      curl_easy_cleanup(curl);
-      curl_global_cleanup();
-
-      // Return success.
-      return true;
-    }
-
+  /*  The iSENSE API gives us two response codes to check against:
+   *  Success: 200 OK
+   *  Failure: 401 Unauthorized
+   */
+  if (http_code == 200) {
     // Clean up cURL
     curl_easy_cleanup(curl);
     curl_global_cleanup();
+
+    // Return success.
+    return true;
   }
 
+  // Clean up cURL
+  curl_easy_cleanup(curl);
+  curl_global_cleanup();
+
   // If curl fails, return false.
   return false;
 }
 
 
-// This is pretty much straight from the GET_curl.cpp file.
-bool iSENSE::get_project_fields()
-{
-  // Detect errors. We need a valid project ID before we try and perform a GET request.
-  if(project_ID == "empty" || project_ID.empty())
-  {
-    cerr << "Error - project ID not set!\n";
+// GET the project fields for a given project ID
+bool iSENSE::get_project_fields() {
+  if (project_ID == EMPTY || project_ID.empty()) {

```

Comment:
Like you mentioned in your comment for this pull request, it would be nice to see a single `isEmpty(id)` type of method in the future.  Not in the scope of this pull request, but just reiterating.